src/server/pkg/obj/google_client.go

```diff
@@ -47,9 +47,12 @@ func (c *googleClient) Walk(name string, fn func(name string) error) error {
 	return nil
 }
 
-//TODO size 0 means read all
+//size 0 means read all
 func (c *googleClient) Reader(name string, offset uint64, size uint64) (io.ReadCloser, error) {
-	return c.bucket.Object(name).NewReader(c.ctx)
+	if size == 0 {
+		return c.bucket.Object(name).NewReader(c.ctx)
+	}
+	return c.bucket.Object(name).NewRangeReader(c.ctx, int64(offset), int64(size))
```

Comment:
Could you check if the aws impl has this behavior implemented as well? I'd like to keep them in sync as much as possible.